sanssouci/lambda_calibration.py

```diff
@@ -179,3 +179,94 @@ def get_pivotal_stats(p0, inverse_template=inverse_linear_template, K=-1):
     pivotal_stats = np.min(tk_inv_all[:, :K], axis=1)
 
     return pivotal_stats
+
+
+def get_pivotal_stats_min(
+    p0, inverse_template=inverse_linear_template,
+    Kmin=0, K=-1
+):
+    """Get pivotal statistic
+
+    Parameters
+    ----------
+
+    p0 :  array-like of shape (B, p)
+        A numpy array of size [B,p] of null p-values obtained from
+        B permutations for p hypotheses.
+    inverse_template : function
+        A function with the same I/O as inverse_template_linear
+    K :  int
+        For JER control over 1:K, i.e. joint control of all k-FWER, k<= K.
+        Automatically set to p if its input value is < 0.
+
+    Returns
+    -------
+
+    array-like of shape (B,)
+        A numpy array of of size [B]  containing the pivotal statitics, whose
+        j-th entry corresponds to \psi(g_j.X) with notation of the AoS 2020
+        paper cited below (section 4.5) [1]_
+
+    References
+    ----------
+
+    .. [1] Blanchard, G., Neuvial, P., & Roquain, E. (2020). Post hoc
+        confidence bounds on false positives using reference families.
+        Annals of Statistics, 48(3), 1281-1303.
+    """
+    # Sort permuted p-values
+    p0 = np.sort(p0, axis=1)
+
+    # Step 3: apply template function
+    # For each feature p, compare sorted permuted p-values to template
+    B, p = p0.shape
+    tk_inv_all = np.array([inverse_template(p0[:, i], i + 1, p)
+                           for i in range(p)]).T
+
+    if K < 0:
+        K = tk_inv_all.shape[1]  # tkInv_all.shape[1] is equal to p
+
+    # Step 4: report min for each row
+    pivotal_stats = np.min(tk_inv_all[:, Kmin:K], axis=1)
+
+    return pivotal_stats
+
+
+def estimate_jer(template, pval0, Kmin, Kmax):
+
+    """
+    Compute empirical JER for a given template and permuted p-values
+    """
+
+    B, p = pval0.shape
+    id_ranks = np.tile(np.arange(0, p), (B, 1))
+
+    cutoffs = np.searchsorted(template, pval0)
+
+    signs = np.sign(id_ranks - cutoffs)
+    sgn_trunc = signs[:, Kmin: Kmax]
+    JER = np.sum([np.any(sgn_trunc[perm] >= 0) for perm in range(B)]) / B
+
+    return JER
+
+
+def calibrate_jer(alpha, learned_templates, pval0, Kmin, Kmax, min_dist=3):
```

Comment:
If the function is meant to be an internal/ancillary ('private') function, it should be denoted `_calibrate_jer`

Reply:
Kmin -> k_min
Kmax -> k_max
(use underscore not camelcase)